### PR TITLE
Add circleci deployment steps

### PR DESCRIPTION
**Pull request by mrfyda**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -9,3 +9,13 @@ test:
     - nvm install 7.4.0 && npm run test
   post:
     - chmod +x ./bin/codacy-coverage.js && cat ./coverage/lcov.info | node ./bin/codacy-coverage.js
+
+deployment:
+  npm:
+    branch: master
+    commands:
+      - echo -e "_auth = $NPM_TOKEN\nemail = $NPM_EMAIL" > .npmrc
+      - npm version minor -m "Release version %s"
```
Comment (rtfpessoa) [inline circle.yml]: Why are we releasing minor versions by default?

File: circle.yml [hunk 2]
```diff
@@ -9,3 +9,13 @@ test:
     - nvm install 7.4.0 && npm run test
   post:
     - chmod +x ./bin/codacy-coverage.js && cat ./coverage/lcov.info | node ./bin/codacy-coverage.js
+
+deployment:
```
Comment (rtfpessoa) [inline circle.yml]: Isn't there any other way for us to opt in for a release? If we merge 3 commits. It does not make much sense to release three versions.

File: circle.yml [hunk 3]
```diff
@@ -9,3 +9,13 @@ test:
...
+  npm:
+    branch: master
+    commands:
+      - echo -e "_auth = $NPM_TOKEN\nemail = $NPM_EMAIL" > .npmrc
+      - npm version minor -m "Release version %s"
+      - git push --tags
```
Comment (machadoit) [inline circle.yml]: I don't think that we can push tags from circleCI as it is (the deploy key seems to be read-only)

**Comment by MAGALY956** [comment]
https://www.apache.org/licenses/

**Comment by rtfpessoa** [comment]
We need to think a bit more about this in the future.

**Pull request closed without merging**